warehouse/warehouse.go

```diff
@@ -217,269 +274,224 @@ func setupTables(dbHandle *sql.DB) error {
+		a.logger.Warnf("retrying warehouse database migration in %s: %v", t, err)
 	})
 	if err != nil {
-		return fmt.Errorf("could not run warehouse database migrations: %w", err)
+		return fmt.Errorf("could not migrate: %w", err)
 	}
-	return nil
-}
 
-func isUploadTriggered(wh model.Warehouse) bool {
-	triggerUploadsMapLock.RLock()
-	defer triggerUploadsMapLock.RUnlock()
-	return triggerUploadsMap[wh.Identifier]
-}
-
-func triggerUpload(wh model.Warehouse) {
-	triggerUploadsMapLock.Lock()
-	defer triggerUploadsMapLock.Unlock()
-	pkgLogger.Infof("[WH]: Upload triggered for warehouse '%s'", wh.Identifier)
-	triggerUploadsMap[wh.Identifier] = true
-}
-
-func clearTriggeredUpload(wh model.Warehouse) {
-	triggerUploadsMapLock.Lock()
-	defer triggerUploadsMapLock.Unlock()
-	delete(triggerUploadsMap, wh.Identifier)
-}
-
-func getConnectionString() string {
-	if !CheckForWarehouseEnvVars() {
-		return misc.GetConnectionString(config.Default)
-	}
-	return fmt.Sprintf("host=%s port=%d user=%s "+
-		"password=%s dbname=%s sslmode=%s application_name=%s",
-		host, port, user, password, dbname, sslMode, appName)
-}
-
-// CheckForWarehouseEnvVars Checks if all the required Env Variables for Warehouse are present
-func CheckForWarehouseEnvVars() bool {
-	return config.IsSet("WAREHOUSE_JOBS_DB_HOST") &&
-		config.IsSet("WAREHOUSE_JOBS_DB_USER") &&
-		config.IsSet("WAREHOUSE_JOBS_DB_DB_NAME") &&
-		config.IsSet("WAREHOUSE_JOBS_DB_PASSWORD")
-}
-
-func setupDB(ctx context.Context, connInfo string) error {
-	var err error
-	dbHandle, err = sql.Open("postgres", connInfo)
-	if err != nil {
-		return err
-	}
-	dbHandle.SetMaxOpenConns(config.GetInt("Warehouse.maxOpenConnections", 20))
-
-	isDBCompatible, err := validators.IsPostgresCompatible(ctx, dbHandle)
-	if err != nil {
-		return err
-	}
-
-	if !isDBCompatible {
-		err := errors.New("rudder Warehouse Service needs postgres version >= 10. Exiting")
-		pkgLogger.Error(err)
-		return err
-	}
-
-	if err = dbHandle.PingContext(ctx); err != nil {
-		return fmt.Errorf("could not ping WH db: %w", err)
-	}
-
-	wrappedDBHandle = sqlquerywrapper.New(
-		dbHandle,
-		sqlquerywrapper.WithLogger(pkgLogger.Child("dbHandle")),
-		sqlquerywrapper.WithQueryTimeout(dbHandleTimeout),
-	)
-
-	return setupTables(dbHandle)
-}
-
-// Setup prepares the database connection for warehouse service, verifies database compatibility and creates the required tables
-func Setup(ctx context.Context) error {
-	if !db.IsNormalMode() {
-		return nil
-	}
-	psqlInfo := getConnectionString()
-	if err := setupDB(ctx, psqlInfo); err != nil {
-		return fmt.Errorf("cannot setup warehouse db: %w", err)
-	}
 	return nil
 }
 
 // Start starts the warehouse service
-func Start(ctx context.Context, app app.App) error {
-	application = app
-
-	mode := config.GetString("Warehouse.mode", config.EmbeddedMode)
-
-	if dbHandle == nil && !isStandAloneSlave(mode) {
-		return errors.New("warehouse service cannot start, database connection is not setup")
-	}
```

Comment:
No point in throwing an error, since later we are already skipping the start for the warehouse service.